### ISSUE TITLE
API Mismatch: Invalid call to '__iadd__()' in googlenet-cifar10/train.mojo

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ProjectOdyssey#2869]
## API Mismatch: Invalid call to '__iadd__()'

**Category**: API Bug
**Priority**: High
**Component**: examples/googlenet-cifar10

### Affected File
- `/home/mvillmow/ml-odyssey/examples/googlenet-cifar10/train.mojo` (line 320)

### Error
```
error: invalid call to '__iadd__': right side cannot be converted from 'Float32' to 'ExTensor'
```

### Root Cause
The code attempts to use the `+=` operator (which calls `__iadd__()`) to add a scalar `Float32` value (loss) to `total_loss`. However, `ExTensor.__iadd__()` requires an `ExTensor` operand, not a scalar.

### Required Fix
- [ ] Use scalar addition: `total_loss = total_loss + loss` instead of `total_loss += loss`
- [ ] Or convert scalar to ExTensor before accumulation
- [ ] Add unit tests for loss accumulation
- [ ] Verify compilation passes: `pixi run mojo build -I . examples/googlenet-cifar10/train.mojo`

### Acceptance Criteria
- [ ] File compiles successfully without errors
- [ ] Loss accumulation works correctly
- [ ] Training loop produces expected loss values
